browser/ai_chat/ai_chat_agent_profile_manager.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_AI_CHAT_AI_CHAT_AGENT_PROFILE_MANAGER_H_
+#define BRAVE_BROWSER_AI_CHAT_AI_CHAT_AGENT_PROFILE_MANAGER_H_
+
+#include "base/files/file_path.h"
+#include "brave/components/ai_chat/core/common/buildflags/buildflags.h"
+#include "chrome/browser/profiles/profile_attributes_storage.h"
+#include "chrome/browser/profiles/profile_manager_observer.h"
+
+static_assert(BUILDFLAG(ENABLE_BRAVE_AI_CHAT_AGENT_PROFILE));
+
+class ProfileManager;
+
+namespace ai_chat {
+
+// Observes the profile system and manages state for any
+// created AI Chat agent profiles.
+// Also observes any browser created and ensures the AI Chat side
+// panel is initially opened.
+class AIChatAgentProfileManager : public ProfileAttributesStorage::Observer,
```

Comment:
AIChatAgentProfileObserver? AIChatAgentProfileInitializer? Not a big deal, I just don't want this to be confused with the actual ProfileManager

Reply:
`GlicProfileManager` does quite a bit more though and again I think this naming makes more sense if we move the functions like `OpenBrowserWindowForAIChatAgentProfile` into this class. Not a blocking issue

Reply:
agreed it can move here. I added a todo in the code https://github.com/brave/brave-browser/issues/48190